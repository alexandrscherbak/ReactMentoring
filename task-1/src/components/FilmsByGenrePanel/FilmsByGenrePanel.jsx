import React from 'react';

const FilmsByGenrePanel = ({genre}) => (
	<div>
		<span>Films by {genre} genre</span>

		<style jsx>
			{`
				div {
					background-color: #f5f5f5;
					color: #919191;
					font-weight: bold;
					height: 50px;
					line-height: 50px;
					padding: 0 100px;
				}
			`}
		</style>
	</div>
);

export default FilmsByGenrePanel;
