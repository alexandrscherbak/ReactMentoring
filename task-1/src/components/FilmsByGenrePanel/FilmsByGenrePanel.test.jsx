import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FilmsByGenrePanel from './FilmsByGenrePanel.jsx';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

const props = {
	genre: 'action',
};

function setup() {
	return mount(<FilmsByGenrePanel {...props}/>);
}

describe('FilmsByGenrePanel', () => {
	test('snapshot', () => {
		const component = renderer.create(
			<FilmsByGenrePanel {...props}/>,
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should render self and subcomponents', () => {
		const enzymeWrapper = setup();
		expect(enzymeWrapper.find('span').text()).toBe(`Films by ${props.genre} genre`);
	});
});
