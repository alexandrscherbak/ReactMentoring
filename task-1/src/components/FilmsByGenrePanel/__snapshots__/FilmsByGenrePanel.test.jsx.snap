// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsByGenrePanel snapshot 1`] = `
<div
  className="jsx-4002406560"
>
  <span
    className="jsx-4002406560"
  >
    Films by 
    action
     genre
  </span>
</div>
`;
